rslib/src/backend/mod.rs

```diff
@@ -137,10 +139,39 @@ impl Backend {
             .clone()
     }
 
-    fn web_client(&self) -> &Client {
-        // currently limited to http1, as nginx doesn't support http2 proxies
-        self.web_client
-            .get_or_init(|| Client::builder().http1_only().build().unwrap())
+    #[cfg(feature = "rustls")]
+    fn set_custom_cert(&self, cert_str: String) -> Result<()> {
+        if let Ok(certificate) = Certificate::from_pem(cert_str.as_bytes()) {
+            let mut client_mutex = self.web_client.lock();
+
+            if let Ok(ref mut web_client) = client_mutex {
+                let _ = (**web_client).insert(
+                    Client::builder()
+                        .use_rustls_tls()
+                        .add_root_certificate(certificate)
+                        .http1_only()
+                        .build()
+                        .unwrap(),
+                );
+                return Ok(());
+            } else {
+                return Err(AnkiError::Interrupted);
+            }
+        }
+
+        Err(AnkiError::InvalidCertificateFormat)
+    }
+
+    fn web_client(&self) -> Result<Client> {
+        let mut client_mutex = self.web_client.lock();
+
+        if let Ok(ref mut web_client) = client_mutex {
+            return Ok((**web_client)
+                .get_or_insert(Client::builder().http1_only().build().unwrap())
+                .clone());
+        }
+
+        Err(AnkiError::Interrupted)
```

Comment:
Does this even need a fallible return value? The only thing being checked is the return value of lock(), which should be unwrapped. If you make this just return the client, then you shouldn't even need to bubble the error up.

Also, you're creating a client even if there is an existing one, and then discarding the value. The build() call should only be made if the option is empty.